### DISCUSSION TITLE
How to use Javascript library update() function?

### DISCUSSION THREAD

**Discussion by jhanstra** [supabase/supabase#13438] [Questions] [answered]
Hi there. It says in the documentation that I should be able to target updating only a certain property within a JSONB object by specifying only that object, like this:
```
const newData = { 'fields': { 'title': 'My new title' } }
const idToUpdate = 'some-id'
const { data, error } = await supabase.from('my-table').update(newData).eq('id', idToUpdate).select()
```

However this does not work for me and instead rewrites everything in the fields object to just { title: 'My new title' }.
How can I get Supabase to only update certain fields?

**Comment by GaryAustin1** [comment, marked as answer]
I don't believe that is possible.  Where are you seeing that you can update individual parts of a jsonb column?
![image](https://user-images.githubusercontent.com/54564956/229969248-7fecf242-b555-42d4-97e9-e6b51a65b254.png)

**Reply by jhanstra** [reply]
Dang, yea I missed that. Thanks for the answer. I guess I just assumed from the example on https://supabase.com/docs/reference/javascript/update that they were updating just the address, since it didn't have any other data attached. Good to know that's how it worked but it would be super useful if there was another method to only update what you send it (or an option on the update method).
